cpp/src/join/hash_join.cu

```diff
@@ -132,8 +132,11 @@ probe_join_hash_table(cudf::table_device_view build_table,
   constexpr cudf::detail::join_kind ProbeJoinKind = (JoinKind == cudf::detail::join_kind::FULL_JOIN)
                                                       ? cudf::detail::join_kind::LEFT_JOIN
                                                       : JoinKind;
-  std::size_t const join_size = output_size.value_or(compute_join_output_size<ProbeJoinKind>(
-    build_table, probe_table, hash_table, compare_nulls, stream));
+
+  size_t const join_size = output_size
```

Comment:
```suggestion
  std::size_t const join_size = output_size
```

Reply:
I prefer `auto const` ;)

Reply:
that's fine too, but the namespace shouldn't have been dropped.

Reply:
Woops. Thanks, will fix

Reply:
brought back `std::`: https://github.com/rapidsai/cudf/pull/9649/commits/74a762f0c34892a250a1570321e4ec69b5394e76